nx/source/services/pcv.c

```diff
@@ -104,3 +104,75 @@ Result pcvGetClockRate(PcvModule module, u32 *out_hz) {
 
     return rc;
 }
+
+Result pcvSetVoltageEnabled(bool state, u32 voltage) {
+    IpcCommand c;
+    ipcInitialize(&c);
+
+    struct {
+        u64 magic;
+        u64 cmd_id;
+        u8 state;
+        u32 voltage;
+    } *raw;
+
+    raw = serviceIpcPrepareHeader(&g_pcvSrv, &c, sizeof(*raw));
+
+    raw->magic = SFCI_MAGIC;
+    raw->cmd_id = 8;
+    raw->state = (u8)state;
+    raw->voltage = voltage;
+
+    Result rc = serviceIpcDispatch(&g_pcvSrv);
+
+    if (R_SUCCEEDED(rc)) {
+        IpcParsedCommand r;
+        struct {
+            u64 magic;
+            u64 result;
+        } *resp;
+
+        serviceIpcParse(&g_pcvSrv, &r, sizeof(*resp));
+        resp = r.Raw;
+
+        rc = resp->result;
+    }
+
+    return rc;
+}
+
+Result pcvGetVoltageEnabled(bool *isEnabled, u32 voltage) {
+    IpcCommand c;
+    ipcInitialize(&c);
+
+    struct {
+        u64 magic;
+        u64 cmd_id;
+        u32 voltage;
+    } *raw;
+
+    raw = serviceIpcPrepareHeader(&g_pcvSrv, &c, sizeof(*raw));
+
+    raw->magic = SFCI_MAGIC;
+    raw->cmd_id = 9;
+    raw->voltage = voltage;
+
+    Result rc = serviceIpcDispatch(&g_pcvSrv);
+
+    if (R_SUCCEEDED(rc)) {
+        IpcParsedCommand r;
+        struct {
+            u64 magic;
+            u64 result;
+            u8 isEnabled;
+        } *resp;
+
+        serviceIpcParse(&g_pcvSrv, &r, sizeof(*resp));
+        resp = r.Raw;
+
+        rc = resp->result;
+        *isEnabled = (bool)resp->isEnabled;
```

Comment:
if (R_SUCCEEDED(rc) && isEnabled) ...

Reply:
Might want to do something similar with out_hz above too

Reply:
sure